pkg/apis/networking/metadata_validation_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package networking
+
+import (
+	"context"
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+)
+
+func TestValidateObjectMetadata(t *testing.T) {
+	cases := []struct {
+		name       string
+		objectMeta metav1.Object
+		ctx        context.Context
+		expectErr  *apis.FieldError
+	}{{
+		name: "invalid knative prefix annotation",
+		objectMeta: &metav1.ObjectMeta{
+			GenerateName: "some-name",
+			Annotations: map[string]string{
+				"networking.knative.dev/testAnnotation": "value",
+			},
+		},
+		expectErr: apis.ErrInvalidKeyName("networking.knative.dev/testAnnotation", apis.CurrentField),
+	}, {
+		name: "valid non-knative prefix annotation label",
```

Comment:
`annotation label` sounds strange. You wanted to say "annotation key"?

Reply:
Done